src/main/java/randoop/test/RegressionCaptureGenerator.java

```diff
@@ -187,21 +196,27 @@ public RegressionChecks generateTestChecks(ExecutableSequence sequence) {
               checks.add(new ObjectCheck(new IsNotNull(), var));
             }
 
-            // Put out any observers that exist for this type
+            // Put out any side-effect-free methods that exist for this type
             Variable var0 = sequence.sequence.getVariable(i);
-            Set<TypedOperation> observers = observerMap.getValues(var0.getType());
-            if (observers != null) {
-              for (TypedOperation m : observers) {
-
-                // When outputting checks, ignore observers that don't take a single argument.
+            Set<TypedClassOperation> sideEffectFreeMethods =
+                sideEffectFreeMap.getValues(var0.getType());
+            if (sideEffectFreeMethods != null) {
+              for (TypedClassOperation m : sideEffectFreeMethods) {
+                // When outputting checks, ignore side-effect-free methods that don't take a single
+                // argument.
                 if (m.getInputTypes().size() != 1) {
```

Comment:
There is code duplication with code in GenTests.  Abstract it out for clarity, brevity, and to reduce the chance of errors.